src/wav2vec2/modeling.py

```diff
@@ -0,0 +1,311 @@
+# __author__ = "Vasudev Gupta"
+# __author_email__ = "7vasudevgupta@gmail.com"
+
+"""TensorFlow implementation of Wav2Vec2"""
+
+import os
+import subprocess
+from dataclasses import replace
+
+import numpy as np
+import tensorflow as tf
+from huggingface_hub import ModelHubMixin
+
+from .config import Wav2Vec2Config
+from .tensorflow_addons import Conv1DWithWeightNorm, GroupNormalization
+
+
+class TransformerAttention(tf.keras.layers.Layer):
+    """Attention layer from `Attention Is All You Need`"""
+
+    def __init__(self, config, name="attention"):
+        super().__init__(name=name)
+        self.num_heads = config.num_heads
+
+        self.q = tf.keras.layers.Dense(config.hidden_size, name="q_proj")
+        self.k = tf.keras.layers.Dense(config.hidden_size, name="k_proj")
+        self.v = tf.keras.layers.Dense(config.hidden_size, name="v_proj")
+        self.projection = tf.keras.layers.Dense(config.hidden_size, name="out_proj")
+
+        self.dropout = tf.keras.layers.Dropout(config.dropout)
+
+    def call(self, batch, training=False):
```

Comment:
Please add a docstring to the public methods here so it's clear what their inputs and outputs are